FirebaseFunctions/Tests/SwiftIntegration/IntegrationTests.swift

```diff
@@ -0,0 +1,258 @@
+        XCTAssertEqual(long, 420)
+        XCTAssertEqual(code, 42)
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result")
+      }
+    }
+    waitForExpectations(timeout: 1)
+  }
+
+  func testScalar() {
+    let expectation = self.expectation(description: #function)
+    let function = functions.httpsCallable("scalarTest")
+    XCTAssertNotNil(function)
+    function.call(17 as Int16) { result, error in
+      do {
+        XCTAssertNil(error)
+        let data = try XCTUnwrap(result?.data as? Int)
+        XCTAssertEqual(data, 76)
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result")
+      }
+    }
+    waitForExpectations(timeout: 1)
+  }
+
+  func testToken() {
+    // Recreate _functions with a token.
+    let functions = FunctionsFake(
+      projectID: "functions-integration-test",
+      region: "us-central1",
+      customDomain: nil,
+      withToken: "token"
+    )
+    functions.useLocalhost()
+
+    let expectation = self.expectation(description: #function)
+    let function = functions.httpsCallable("FCMTokenTest")
+    XCTAssertNotNil(function)
+    function.call([:]) { result, error in
+      do {
+        XCTAssertNil(error)
+        let data = try XCTUnwrap(result?.data) as? [String: Int]
+        XCTAssertEqual(data, [:])
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result")
+      }
+    }
+    waitForExpectations(timeout: 1)
+  }
+
+  func testFCMToken() {
+    let expectation = self.expectation(description: #function)
+    let function = functions.httpsCallable("FCMTokenTest")
+    XCTAssertNotNil(function)
+    function.call([:]) { result, error in
+      do {
+        XCTAssertNil(error)
+        let data = try XCTUnwrap(result?.data) as? [String: Int]
+        XCTAssertEqual(data, [:])
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result")
+      }
+    }
+    waitForExpectations(timeout: 1)
+  }
+
+  func testNull() {
+    let expectation = self.expectation(description: #function)
+    let function = functions.httpsCallable("nullTest")
+    XCTAssertNotNil(function)
+    function.call(nil) { result, error in
+      do {
+        XCTAssertNil(error)
+        let data = try XCTUnwrap(result?.data) as? NSNull
+        XCTAssertEqual(data, NSNull())
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result")
+      }
+    }
+    waitForExpectations(timeout: 1)
+  }
+
+  func testMissingResult() {
+    let expectation = self.expectation(description: #function)
+    let function = functions.httpsCallable("missingResultTest")
+    XCTAssertNotNil(function)
+    function.call(nil) { result, error in
+      do {
+        XCTAssertNotNil(error)
+        let error = try XCTUnwrap(error! as NSError)
```

Comment:
I wonder what `!` is needed for in this line?

Reply:
The paren's were in the wrong place. Fixed. Good catch!